[Bug]: Parquet File Output fails in Hop 2.16.0 with NoClassDefFoundError (worked in 2.15.0)

### Apache Hop version?

2.16.0

### Java version?

17.0.17

### Operating system

Docker

### What happened?

When upgrading from Apache Hop 2.15.0 to Apache Hop 2.16.0, the Parquet File Output transform stops working.
The same pipeline and configuration that previously worked now fails with a NoClassDefFoundError related to a missing Woodstox class:
com.ctc.wstx.io.InputBootstrapper.

This seems to be a dependency regression, the Parquet output transform requires a Woodstox/WSTX XML parser class that is no longer available in the classpath in version 2.16.0.

**Environment**

Hop version: 1.16.0

Working version: 2.15.0
Execution mode: Hop Web / Docker
Transform: Parquet File Output

**Steps to Reproduce**

Create a pipeline using Parquet File Output
Run the pipeline on Hop 2.16.0
Observe that Hop fails when trying to write the parquet file

**Error logs:** 
ERROR: Unexpected error
ERROR: java.lang.NoClassDefFoundError: com/ctc/wstx/io/InputBootstrapper
    at org.apache.hop.parquet.transforms.output.ParquetOutput.openNewFile(ParquetOutput.java:135)
    at org.apache.hop.parquet.transforms.output.ParquetOutput.processRow(ParquetOutput.java:100)
    at org.apache.hop.pipeline.transform.RunThread.run(RunThread.java:54)
    at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.ClassNotFoundException: com.ctc.wstx.io.InputBootstrapper
    at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1377)
    at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1186)


### Issue Priority

Priority: 3

### Issue Component

Component: Hop Web

--- a/engine-beam/pom.xml
+++ b/engine-beam/pom.xml
@@ -70,6 +70,10 @@
                     <artifactId>beam-sdks-java-transform-service-launcher</artifactId>
                 </exclusion>
                 <exclusion>
+                    <groupId>org.apache.commons</groupId>
+                    <artifactId>commons-lang3</artifactId>
+                </exclusion>
+                <exclusion>
                     <groupId>org.checkerframework</groupId>
                     <artifactId>checker-qual</artifactId>
                 </exclusion>

--- a/lib/pom.xml
+++ b/lib/pom.xml
@@ -140,7 +140,7 @@
         <springframework4.version>4.3.18</springframework4.version>
         <stax2-api.version>4.2.2</stax2-api.version>
         <tyrus-standalone-client.version>1.13.1</tyrus-standalone-client.version>
-        <woodstox.version>6.4.0</woodstox.version>
+        <woodstox.version>7.1.1</woodstox.version>
         <xerces.version>2.12.2</xerces.version>
         <xml-apis-ext.version>1.3.04</xml-apis-ext.version>
         <xmlunit.version>1.5</xmlunit.version>
@@ -818,6 +818,11 @@
                 <groupId>org.jsoup</groupId>
                 <artifactId>jsoup</artifactId>
                 <version>${jsoup.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.fasterxml.woodstox</groupId>
+                <artifactId>woodstox-core</artifactId>
+                <version>${woodstox.version}</version>
             </dependency>
         </dependencies>
     </dependencyManagement>

--- a/plugins/engines/beam/src/assembly/assembly.xml
+++ b/plugins/engines/beam/src/assembly/assembly.xml
@@ -154,6 +154,9 @@
                 <exclude>com.google.code.gson:gson:jar</exclude>
                 <exclude>org.apache.commons:commons-collections4:jar</exclude>
                 <exclude>org.jspecify:jspecify:jar</exclude>
+                <exclude>org.apache.commons:commons-lang3:jar</exclude>
+                <exclude>commons-collections:commons-collections:jar</exclude>
+                <exclude>com.fasterxml.woodstox:woodstox-core:jar</exclude>
             </excludes>
             <outputDirectory>lib/beam</outputDirectory>
         </dependencySet>
@@ -252,6 +255,9 @@
                 <include>io.confluent::jar</include>
                 <include>com.google.code.gson:gson:jar</include>
                 <include>org.apache.commons:commons-collections4:jar</include>
+                <include>org.apache.commons:commons-lang3:jar</include>
+                <include>commons-collections:commons-collections:jar</include>
+                <include>com.fasterxml.woodstox:woodstox-core:jar</include>
             </includes>
             <outputDirectory>lib/core</outputDirectory>
         </dependencySet>

--- a/plugins/tech/neo4j/src/assembly/assembly.xml
+++ b/plugins/tech/neo4j/src/assembly/assembly.xml
@@ -51,6 +51,7 @@
             <excludes>
                 <exclude>org.apache.hop:hop-tech-neo4j:jar</exclude>
                 <exclude>org.reactivestreams:reactive-streams:jar</exclude>
+                <exclude>io.projectreactor:reactor-core:jar</exclude>
                 <exclude>io.netty::jar</exclude>
             </excludes>
             <outputDirectory>plugins/tech/neo4j/lib</outputDirectory>
@@ -59,6 +60,7 @@
             <scope>runtime</scope>
             <includes>
                 <include>org.reactivestreams:reactive-streams:jar</include>
+                <include>io.projectreactor:reactor-core:jar</include>
                 <include>io.netty::jar</include>
             </includes>
             <outputDirectory>lib/core</outputDirectory>

--- a/plugins/tech/parquet/pom.xml
+++ b/plugins/tech/parquet/pom.xml
@@ -29,7 +29,7 @@
     <name>Hop Plugins Technology Parquet</name>
 
     <properties>
-        <parquet.version>1.15.2</parquet.version>
+        <parquet.version>1.16.0</parquet.version>
     </properties>
 
     <dependencyManagement>
@@ -45,6 +45,10 @@
     </dependencyManagement>
 
     <dependencies>
+        <dependency>
+            <groupId>com.fasterxml.woodstox</groupId>
+            <artifactId>woodstox-core</artifactId>
+        </dependency>
         <dependency>
             <groupId>org.apache.hadoop</groupId>
             <artifactId>hadoop-client</artifactId>
@@ -104,21 +108,6 @@
                     <artifactId>snappy-java</artifactId>
                 </exclusion>
             </exclusions>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.parquet</groupId>
-            <artifactId>parquet-format-structures</artifactId>
-            <version>${parquet.version}</version>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.commons</groupId>
-            <artifactId>commons-lang3</artifactId>
-            <scope>provided</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.commons</groupId>
-            <artifactId>commons-math3</artifactId>
-            <scope>provided</scope>
         </dependency>
     </dependencies>
 </project>

--- a/plugins/tech/parquet/src/assembly/assembly.xml
+++ b/plugins/tech/parquet/src/assembly/assembly.xml
@@ -48,12 +48,12 @@
         <dependencySet>
             <scope>runtime</scope>
             <excludes>
-                <exclude>
-                    org.apache.hop:hop-tech-parquet:jar
-                </exclude>
+                <exclude>org.apache.hop:hop-tech-parquet:jar</exclude>
+                <exclude>com.fasterxml.woodstox:woodstox-core:jar</exclude>
             </excludes>
             <includes>
                 <include>org.apache.parquet:*:jar</include>
+                <include>org.locationtech.jts:jts-core:jar</include>
             </includes>
             <outputDirectory>plugins/tech/parquet/lib</outputDirectory>
         </dependencySet>
@@ -62,6 +62,7 @@
             <includes>
                 <include>org.apache.hadoop:*:jar</include>
                 <include>org.apache.hadoop.thirdparty:*:jar</include>
+                <include>com.fasterxml.woodstox:woodstox-core:jar</include>
             </includes>
             <outputDirectory>lib/core</outputDirectory>
         </dependencySet>

